tpmutil/mssim/mssim.go

```diff
@@ -0,0 +1,227 @@
+// Package mssim implements the Microsoft simulator TPM2 Transmission Interface
+//
+// The Microsoft simulator TPM Command Transmission Interface (TCTI) is a
+// remote procedure interface donated to the TPM2 Specification by Microsoft.
+// Its primary implementation is the tpm_server maintained by IBM.
+//
+// https://sourceforge.net/projects/ibmswtpm2/
+//
+// This package implements client code to communicate with server code described
+// in the document "TPM2 Specification Part 4: Supporting Routines – Code"
+//
+// https://trustedcomputinggroup.org/wp-content/uploads/TPM-Rev-2.0-Part-4-Supporting-Routines-01.38-code.pdf
+package mssim
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"sync/atomic"
+)
+
+// Constants defined in "D.3.2. Typedefs and Defines"
+const (
+	tpmSignalPowerOn  = 1
+	tpmSignalPowerOff = 2
+	tpmSendCommand    = 8
+	tpmSignalNVOn     = 11
+	tpmSessionEnd     = 20
+)
+
+// Config holds configuration parameters for connecting to the simluator.
+type Config struct {
+	// Addresses of the command and platform handlers.
+	//
+	// Defaults to port 2321 and 2322.
+	CommandAddress  string
+	PlatformAddress string
+
+	// DialContext is used to connect to the simluator ports.
+	//
+	// If nil, a reasonable default is provided.
+	DialContext func(ctx context.Context, network, address string) (net.Conn, error)
```

Comment:
What's the use case for this?
Let's not add abstraction unless necessary.

Reply:
removed